src/io/message_conversion.hpp

```diff
@@ -199,18 +198,24 @@ class OpaquePromise {
 };
 
 struct DeadlineAndOpaquePromise {
+  Time requested_at;
   Time deadline;
   OpaquePromise promise;
 };
 
 template <class From>
-std::type_info const &type_info_for_variant(From const &from) {
+std::type_info const &TypeInfoForVariant(From const &from) {
   return std::visit([](auto &&x) -> decltype(auto) { return typeid(x); }, from);
 }
 
+template <class T>
+utils::TypeInfoRef TypeInfoFor(const T & /* t */) {
+  return typeid(T);
+}
```

Comment:
Is this necessary?

Reply:
I haven't figured out a great way to arrange this, but this method is overloaded in raft.hpp in a way that allows more fine-grained type information to be retrieved from inside of variants and rsm responses. If this was the only usage of it, I would agree that it should not be a method of its own, but it basically exists as an interface that higher level messages can overload to result in finer grained metric collection and clearer output.